File: test/before_train/trainset_and_testset/cpp-py/HumanEval_68_pluck.py
from typing import List

def pluck(arr: List[int]) -> List[int]:
    return [x for x in arr if x % 2 == 0]

def check(candidate):
    assert candidate([4, 2, 3]) == [2, 1]
    assert candidate([1, 2, 3]) == [2, 1]
    assert candidate([]) == []
    assert candidate([5, 0, 3, 0, 4, 2]) == [0, 1]
    assert candidate([1, 2, 3, 0, 5, 3]) == [0, 3]
    assert candidate([5, 4, 8, 4, 8]) == [4, 1]
    assert candidate([7, 6, 7, 1]) == [6, 1]
    assert candidate([7, 9, 7, 1]) == []

def test_check():
    check(pluck)

test_check()

File: codes/src-tgt/py-cpp/HumanEval_68_pluck.cpp
### Python version

from typing import List

def pluck(arr: List[int]) -> List[int]:
    if(len(arr) == 0): return []
    evens = list(filter(lambda x: x%2 == 0, arr))
    if(evens == []): return []
    return [min(evens), arr.index(min(evens))]

### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
std::vector<long> pluck(std::vector<long> arr) {
    if(arr.size() == 0) return {};
    std::vector<long> evens;
    for(auto x: arr) if(x%2 == 0) evens.push_back(x);
    if(evens.size() == 0) return {};
    long min_val = *std::min_element(evens.begin(), evens.end());
    return {min_val, std::find(arr.begin(), arr.end(), min_val) - arr.begin()};
}

